### ISSUE TITLE
Enerwave ZWN-RSM2-PLUS child device doesn't register physical toggle change

### ISSUE THREAD

**Issue by moore-cr** [Mariano-Github/Edge-Drivers-Beta#55]
Hi Mariano,

I have several Enerwave ZWN-RSM2-PLUS devices throughout my home, and they all exhibit the same behavior after switching to your beta edge drivers.

Here's what works:
-Control both main and child switch from SmartThings app. Switch state is properly updated.
-Switch state updates for the main switch when manually toggled.

What doesn't work:
-Switch state does not update for the child switch when manually toggled.

I have one Monoprice dual relay installed as well, and it does properly update switch states when manually toggled. I am more than happy to provide logs or do any troubleshooting as necessary. Thank you so much for all the work that you do for this.

Chris

**Comment by Mariano-Github** [comment]
Hi,

If you want to send a log when you change status on switch2 physio?

I won't be able to do anything until the weekend

**Comment by Mariano-Github** [comment]
I need the fingerprints too

**Comment by moore-cr** [comment]
I've attached 2 files. One is logs from toggling via the physical switch, and the other is from toggling via the app. In both cases, I toggled both the main and child switch.

Fingerprint:
DNI: 0x5E
Manufacturer: 0x011A
Product Type: 0x0111
Product ID: 0x0606

[physical-switch-changes.log](https://github.com/Mariano-Github/Edge-Drivers-Beta/files/10579401/physical-switch-changes.log)
[app-switch-changes.log](https://github.com/Mariano-Github/Edge-Drivers-Beta/files/10579402/app-switch-changes.log)

**Comment by Mariano-Github** [comment]
Hi Chris
In the log when you turn on and off the physical device alone, the driver only receives a message from enpoint 1{}, corresponding to the main switch
From endpoint 2, which corresponds to switch1, no message is received from the device.

In the logs when you do on and off from the todp app it looks good

I don't know why the switch1 device doesn't send anything

**Comment by moore-cr** [comment]
Okay. Thanks for checking. I was afraid of that. I will look and see if there's any firmware parameters that I can adjust to change that. I know that it used to work using custom Groovy drivers.

**Comment by moore-cr** [comment]
Hi Mariano,

As an update, I was able to get the child switch status updates working. The Groovy drivers here: https://github.com/erocm123/SmartThingsPublic/tree/master/devicetypes/erocm123 showed that this device requires specific device associations set. After a bit of investigating, I was able to switch to your "Z-Wave Device Config Mc" driver temporarily to specifically set the Multi Channel Device association for Group 1 and Node 1. Once that was set, I switched back to the "Z-Wave Switch and Child Mc" driver and device status updated correctly for physical switch toggles. Hopefully, this info can help someone else.

Thanks so much for all of your work.

Chris

**Comment by Mariano-Github** [comment]
Thanks

Ae you refer to configure parameter 3 to value 1?

**Comment by Mariano-Github** [comment]
Hi Chris
I don't know exactly what you did, I could put it in the driver so that it is configured when installed

**Comment by moore-cr** [comment]
Hi Mariano,

I tried configuring parameter 3 to 1, but that didn't seem to do anything. I also tried single device association groups 2 and 3 to node 1, but that didn't work either. The change that made a difference was setting multi-channel association group 1 to node 1 (my hub). I've attached a screenshot that shows the settings I used using your config driver. Once I got it working on one device, I tried on the other devices in my home and they all work now.

I hope this helps. If you're able to fold that into your driver, that would be fantastic.

Thanks,

Chris

![2023-2-7_18-5-58](https://user-images.githubusercontent.com/36528832/217387184-6dbd693a-d347-4e14-93bf-d6e28b68873c.PNG)
